docs/reference/mapping/types/search-as-you-type.asciidoc

```diff
@@ -0,0 +1,184 @@
+beginning of the input) and infix completion (i.e. matching terms at any
+position within the input) are supported.
+
+When adding a field of this type to a mapping
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "my_field": {
+        "type": "search_as_you_type"
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+This creates the following fields
+
+[horizontal]
+
+`my_field`::
+
+    Analyzed as configured in the mapping. If an analyzer is not configured,
+    the default analyzer for the index is used
+
+`my_field._2gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 2
+
+`my_field._index_prefix`::
+
+    Wraps the analyzer of `my_field._2gram` with an edge ngram token filter
+
+
+The same input text is indexed into each of these fields automatically, with
+their differing analysis chains, when an indexed document has a value for the
+root field `my_field`.
+
+[source,js]
+--------------------------------------------------
+PUT my_index/_doc/1
+{
+  "my_field": "quick red fox lazy brown dog"
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[continued]
+
+Setting a larger `max_shingle_size` in the mapping will create shingle fields
+with larger sized shingles, of 2 up to `max_shingle_size`. The
+`my_field._index_prefix` subfield will always use the analyzer from the
+subfield with the largest shingle size. For example the mapping
+
+[source,js]
+--------------------------------------------------
+PUT my_index
+{
+  "mappings": {
+    "properties": {
+      "my_field": {
+        "type": "search_as_you_type",
+        "max_shingle_size": 4
+      }
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+
+Will create the fields
+
+[horizontal]
+
+`my_field`::
+
+    Analyzed as configured in the mapping. If an analyzer is not configured,
+    the default analyzer for the index is used
+
+`my_field._2gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 2
+
+`my_field._3gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 3
+
+`my_field._4gram`::
+
+    Wraps the analyzer of `my_field` with a shingle token filter of shingle
+    size 4
+
+`my_field._index_prefix`::
+
+    Wraps the analyzer of `my_field._4gram` with an edge ngram token filter
+
+
+[[specific-params]]
+==== Parameters specific to the `search_as_you_type` field
+
+The following parameters are accepted in a mapping for the `search_as_you_type`
+field and are specific to this field type
+
+[horizontal]
+
+`max_shingle_size`::
+
+    The largest shingle size to index the input with and create subfields for,
+    creating one subfield for each shingle size between 2 and
+    `max_shingle_size`. Accepts integer values between 2 (default) and 4
+    inclusive.
+
```

Comment:
Can you add an example query that targets the different shingles. Something like a `match` query with an `and` operator on the root field in a `must` clause and then one `match` query per shingle size on should clauses with operator set to `or` ? This would explain how we expect to use the shingle fields (adds a cheap proximity boost).

Reply:
👍 I added this query

Reply:
I added the support for `phrase` and `phrase_prefix` query in https://github.com/elastic/elasticsearch/pull/35600/commits/9532c19a64b685a89de017354ef599c880b48bfc so I think we can show a simple example with a `match_phrase_prefix` on the root field.